### ISSUE TITLE
Ruff Cannot Recognise a Variable Usage inside typing.cast

### ISSUE THREAD

**Issue by barmanroys** [astral-sh/ruff#17442]
### Summary

[Play ground link](https://play.ruff.rs/7abdedee-b394-4ce3-a5b4-ecc1904fbfa7). 

##### Context
A test method within a `unittest` class. 

##### Code
```python3
from typing import cast, Tuple
...

def test_run(self):
    """Use the sample initial states and action choices for testing."""
    initial_input: int = 1 # Ruff warns this variable is defined but not used 
    actions: Tuple[int, ...] = (199, 213, 111, 333) # Ruff warns this variable is defined but not used 
    result = cast(
        typ=Tuple[Any],
        val=self.interactor.run(llm_input=initial_input, action_sequence=actions),
    )
    self.assertTupleEqual(tuple1=result, tuple2=(1, 0, True, {}))
    logging.info(msg="Multi-round interactor passed testing.")
```

##### Problem
Ruff Cannot recognise the usage of a variable inside `typing.cast`. Even though it _is_ being used. 

##### System Details
* ruff 0.11.5
* uv 0.5.13
* Operating system: Linux mint 22

### Version

0.11.5

**Comment by sharkdp** [comment]
Here's a smaller example that shows the same problem. It seems to depend on the fact that keyword arguments are used for `typing.cast`:
```py
from typing import cast

def f():
    x = 1
    print(cast(typ=int, val=x))
```

**Comment by barmanroys** [comment]
@sharkdp thanks for reducing my example to a minimal variant. 

So, as `typing.cast` accepts keyword arguments, should this behaviour be fixed on the _ruff_-end of things (perdon the pun)?

**Comment by ntBre** [comment]
I think we're only visiting the positional args for `cast` (and the other `typing` imports in this block?):

https://github.com/astral-sh/ruff/blob/d4695feccf28b46883cb1303a811f147b812ab72/crates/ruff_linter/src/checkers/ast/mod.rs#L1541-L1553

**Comment by dhruvmanila** [comment]
> I think we're only visiting the positional args for `cast` (and the other `typing` imports in this block?):

Yeah, this seems like a clear issue. We should probably use `find_argument_value` or `find_argument` which finds an argument via both position and name.
